jira7/jira_endpoint.bal

```diff
@@ -145,90 +144,88 @@ public type Client client object {
+    public remote function getProjectCategory(string projectCategoryId) returns ProjectCategory|error;
 
     # Create a new project category.
     # + newCategory - Record which contains the mandatory fields for new project category creation
-    # + return - `ProjectCategory` type records or `JiraConnectorError` record
-    public remote function createProjectCategory(ProjectCategoryRequest newCategory)
-                               returns ProjectCategory|JiraConnectorError;
+    # + return - Returns `ProjectCategory` record when successful, else returns an error
+    public remote function createProjectCategory(ProjectCategoryRequest newCategory) returns ProjectCategory|error;
 
     # Delete a project category.
     # + projectCategoryId - Jira id of the project category
-    # + return - returns true if the process is successful or `JiraConnectorError` record
-    public remote function deleteProjectCategory(string projectCategoryId) returns boolean|JiraConnectorError;
+    # + return - Returns true when the process is successful, else returns an error
+    public remote function deleteProjectCategory(string projectCategoryId) returns boolean|error;
 
     # Returns a detailed representation of a jira issue.
     # + issueIdOrKey - Id or key of the required issue
-    # + return - `Issue` type record or `JiraConnectorError` record
-    public remote function getIssue(string issueIdOrKey) returns Issue|JiraConnectorError;
+    # + return -  Returns `Issue` record when successful, else returns an error
+    public remote function getIssue(string issueIdOrKey) returns Issue|error;
 
     # Creates a new jira issue.
     # + newIssue - Record which contains the mandatory fields for new issue creation
-    # + return - `Issue` type record or `JiraConnectorError` record
-    public remote function createIssue(IssueRequest newIssue) returns Issue|JiraConnectorError;
+    # + return - Returns `Issue` record when successful, else returns an error
+    public remote function createIssue(IssueRequest newIssue) returns Issue|error;
     # Deletes a jira issue.
     # + issueIdOrKey - Id or key of the issue
-    # + return - returns true if the process is successful or `JiraConnectorError` record
-    public remote function deleteIssue(string issueIdOrKey) returns boolean|JiraConnectorError;
+    # + return - Returns true when the process is successful, else returns an error
+    public remote function deleteIssue(string issueIdOrKey) returns boolean|error;
 
     # Adds a comment to a Jira Issue.
     # + issueIdOrKey - Id or key of the issue
     # + comment - The details of the comment to be added
-    # + return - returns true if the process is successful or `JiraConnectorError` record
-    public remote function addCommentToIssue(string issueIdOrKey, IssueComment comment)
-                               returns boolean|JiraConnectorError;
+    # + return - Returns true when the process is successful, else returns an error
```

Comment:
Do all these functions only return `true`? Never `false`?

Reply:
In the false case it will return an error

Reply:
I think we should be using a finite type with only `true` instead of `boolean` in all these scenarios. 

`type TRUE true;`

then each function would return `TRUE|error` instead of `boolean|error`.

Reply:
Fixed in https://github.com/wso2-ballerina/module-jira/pull/56/commits/53f515c82bfea32761ce3ddc18b489e2d80c67cc

Reply:
On second thoughts, we might not even need `TRUE`. Even though the remote call returns `true` when successful, IMO we could simply represent this as `error?`, where `true` being returned is represented by `()`.

@anupama-pathirage , @pubudu91 WDYT? 

The scenario is that the remote call would return either `true` or `error`. Return options for our remote functions are:
1. `boolean|error` - what we had, kinda redundant since if boolean, always `true`
2. `TRUE|error` - a more accurate representation of the actual possible return values
3. `error?` - where we represent the `true` return by `()` - i.e., not an `error` return